trieste/models/gpflux/models.py

```diff
@@ -143,10 +144,17 @@ def optimize(self, dataset: Dataset) -> None:
         Optimize the model with the specified `dataset`.
         :param dataset: The data with which to optimize the `model`.
         """
-        self.model_keras.fit(
-            {"inputs": dataset.query_points, "targets": dataset.observations}, **self._fit_args
+        fit_args = self._fit_args
+        if fit_args is not None and "epochs" in fit_args:
```

Comment:
I know that that `self._fit_args` is described as `Optional[Dict[str, Any]] ` in DeepGaussianProcess but that isn't right: it's always a dictionary (otherwise **fit_args would fail). If you fix the type declaration you can simplify this:

```suggestion
        if "epochs" in fit_args:
```

Reply:
not sure how I should change this, just remove optional?
`self._fit_args: Dict[str, Any] = {...`